### PR TITLE
Tests and optimize

### PR DESCRIPTION
**Pull request by fschade**
## Description
in #2362 we we started to work intensive on theming. This needs some testing to be sure everything works as expected. In the same breath it was a good moment in time to refactor and align some testing workflows.

once https://github.com/owncloud/web/pull/4822 is finalized we need to touch the theme loading and testing again because the pr introduces theme keys.

## Related Issue

- relates #2362

## Motivation and Context

we need more unit tests

## How Has This Been Tested?

- jest

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests

## Checklist:

- [X] Code changes
- [X] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Changelog

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootEdgeCases</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsEdgeCases</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>TrashbinDelete</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>TrashbinFilesFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingNotifications</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFilePermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersBlacklisted</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRootBlacklisted</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFilePermissionMultipleUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissionMultipleUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>

















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIDelete</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>MarkdownEditor</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRootSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>TrashbinRestore</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersSharingIndicator</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Favorites</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>

























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderAdvPermissionsGrp</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderAdvancedPermissionMU</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpire</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersExpireToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>









































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingAutocompletion</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Move</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Upload</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>













































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionsUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicDifferentRoles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>



















































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>ResharingToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>

























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Resharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>

























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





















































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>


























































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRename</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>







































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsersRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>





























































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>































































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/14189/

<p>











































































































































</p></details>

**Review by LukasHirt — APPROVED** [review]
After bringing back stubs it's good to go 🚀

File: packages/web-runtime/src/helpers/theme.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import defaultTheme from 'web-runtime/themes/owncloud.json'
+export const loadTheme = async (name = '') => {
+  const defaults = { theme: defaultTheme, name: 'themes/owncloud/theme.json' }
```
Comment (kulmann) [inline packages/web-runtime/src/helpers/theme.js:3]: `themes/owncloud/theme.json` doesn't resolve

Comment (kulmann) [reply packages/web-runtime/src/helpers/theme.js:3]: (irrelevant, since we don't use `loadTheme` with an empty name)

Comment (kulmann) [reply packages/web-runtime/src/helpers/theme.js:3]: Ah, thanks for explaining that this name doesn't need to resolve. Only for having a proper key.

Comment (fschade) [reply packages/web-runtime/src/helpers/theme.js:3]: it's just the name, it does not need to resolve

File: changelog/unreleased/change-introduce-web-pkg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Change: Add web-pkg package
```
Comment (kulmann) [inline changelog/unreleased/change-introduce-web-pkg:1]: It's an enhancement (no breaking change => we call non-breaking changes `enhancement`).

**Review by kulmann — APPROVED** [review]

**Pull request merged**